### DISCUSSION TITLE
How to ugrade OpenMage 20.1.x to 20.3.x

### DISCUSSION THREAD

**Discussion by Danieleeffe1** [OpenMage/magento-lts#3713] [General]
I know it seems like a stupid question, but doing a search in the various sections I didn't find an "immediate" answer, therefore in doubt.

How do you update the various versions of openmage?

I'm currently using 20.1.x and would like to upgrade to 20.3.x

I thought it was sufficient to run "composer update" from CMD, but the indicated version remains 20.1.x

Should I download the source zip file, unzip it and overwrite the current version or is there a less brutal way?

Thank you

**Comment by fballiano** [comment]
if you're using composer in your project, you could open the composer.json, change the version there to 20.3.0 and then run a composer install :-)

**Reply by Danieleeffe1** [reply]
Thanks for the reply

I checked in my Composer.json file but I don't see any reference to OpenMage version 20.1.0 to change to 20.3.0

did I do something wrong?

**Reply by fballiano** [reply]
you should have something like

```
"require": {
    "aydin-hassan/magento-core-composer-installer": "2.1.0",
    "magento-hackathon/magento-composer-installer": "4.0.2",
    "cweagans/composer-patches": "1.7.3",
    "openmage/magento-lts": "20.2.0"
  },
```

change "20.2.0" into "20.3.0" and run composer install

**Reply by Danieleeffe1** [reply]
Thanks for the reply

I modified the composer.json file by changing the versions and adding the line

"openmage/magento-lts": "20.3.0",

the system returns this message:

[RuntimeException]
   Root package 'openmage/magento-lts' cannot require itself in its composer.json
   Did you accidentally name your root package after an external package?

if I remove the line

"openmage/magento-lts": "20.3.0",

It works normally as before

**Comment by Flyingmana** [comment]
it depends how you installed OpenMage.

Do you have it as dependency in your composer.json, or is the composer.json you use the one of OpenMage itself?

(the second one you get for example when you downloaded the project via git or a release zip file)

**Reply by Danieleeffe1** [reply]
The second case: I downloaded the OM 20.1.x package and used its composer

**Comment by sreichel** [comment]
Outdated?
